### PR TITLE
Bump golang k8s dependencies to 1.26

### PR DESCRIPTION
**Pull request by pavolloffay**
Related to https://github.com/open-telemetry/opentelemetry-operator/issues/1385

### CODE REVIEW

File: pkg/collector/reconcile/configmap_test.go [hunk 1]
```diff
@@ -103,8 +103,10 @@ receivers:
         metrics_path: /metrics
         scheme: http
         follow_redirects: true
+        enable_http2: true
```
Comment (jaronoff97) [inline pkg/collector/reconcile/configmap_test.go:106]: what are these additions for? new defaults?

**Review by iblancasa — APPROVED** [review]

File: pkg/collector/reconcile/configmap_test.go (hunk 1 above)
Comment (pavolloffay) [reply pkg/collector/reconcile/configmap_test.go:106]: Probably, these changes are coming from the update of `github.com/prometheus`

**Comment by pavolloffay** [comment]
@jaronoff97 @frzifus this PR is ready for review.

**Review by frzifus — APPROVED** [review]

File: pkg/collector/reconcile/configmap_test.go (hunk 1 above)
Comment (pavolloffay) [reply pkg/collector/reconcile/configmap_test.go:106]: @jaronoff97 any concerns or objections to merge this?

Comment (jaronoff97) [reply pkg/collector/reconcile/configmap_test.go:106]: nope, should be okay!

**Pull request merged**